bin/powershell/metric-windows-cpu-load.ps1

```diff
@@ -22,14 +22,40 @@
 #   Copyright 2016 sensu-plugins
 #   Released under the same terms as Sensu (the MIT license); see LICENSE for details.
 #
+
+$ThisProcess = Get-Process -Id $pid
+$ThisProcess.PriorityClass = "BelowNormal"
+
+. (Join-Path $PSScriptRoot perfhelper.ps1)
+
 $ThisProcess = Get-Process -Id $pid
 $ThisProcess.PriorityClass = "BelowNormal"
 
-$Path = hostname
-$Path = $Path.ToLower()
+$perfCategoryID = Get-PerformanceCounterByID -Name 'Processor Information'
+$localizedCategoryName = Get-PerformanceCounterLocalName -ID $perfCategoryID
+
+$perfCounterID = Get-PerformanceCounterByID -Name '% Processor Time'
+$localizedCounterName = Get-PerformanceCounterLocalName -ID $perfCounterID
+$percent_total = [System.Math]::Round((Get-Counter "\$localizedCategoryName(_total)\$localizedCounterName" -SampleInterval 1 -MaxSamples 1).CounterSamples.CookedValue)
+
+$perfCounterID = Get-PerformanceCounterByID -Name '% Idle Time'
+$localizedCounterName = Get-PerformanceCounterLocalName -ID $perfCounterID
+$percent_idle = [System.Math]::Round((Get-Counter "\$localizedCategoryName(_total)\$localizedCounterName" -SampleInterval 1 -MaxSamples 1).CounterSamples.CookedValue)
+
+$perfCounterID = Get-PerformanceCounterByID -Name '% User Time'
+$localizedCounterName = Get-PerformanceCounterLocalName -ID $perfCounterID
+$percent_user = [System.Math]::Round((Get-Counter "\$localizedCategoryName(_total)\$localizedCounterName" -SampleInterval 1 -MaxSamples 1).CounterSamples.CookedValue)
+
+$perfCounterID = Get-PerformanceCounterByID -Name '% Interrupt Time'
+$localizedCounterName = Get-PerformanceCounterLocalName -ID $perfCounterID
+$percent_interrupt = [System.Math]::Round((Get-Counter "\$localizedCategoryName(_total)\$localizedCounterName" -SampleInterval 1 -MaxSamples 1).CounterSamples.CookedValue)
+
 
-$Value = (Get-WmiObject CIM_Processor).LoadPercentage
+$Path = [System.Net.Dns]::GetHostEntry([string]"localhost").HostName.toLower()
 
-$Time = [int][double]::Parse((Get-Date -UFormat %s))
+$Time = DateTimeToUnixTimestamp -DateTime (Get-Date)
```

Comment:
```
-$Path = hostname
-$Path = $Path.ToLower()
-$Value = (Get-WmiObject CIM_Processor).LoadPercentage
 +$Path = [System.Net.Dns]::GetHostEntry([string]"localhost").HostName.toLower()
-$Time = [int][double]::Parse((Get-Date -UFormat %s))
 +$Time = DateTimeToUnixTimestamp -DateTime (Get-Date)
```


This doesn't seem like much of an improvement to me. What where simple oneliners became complicated ones or in the time case even a dependency on another ps1 file. And it does exactly the same in the end?

Reply:
Same goes for every other file where this same thing happens. Seems like over-complicating something super simple to me?

Reply:
Hi simon,

> Have you tested this on multiple different languages btw?

i just tested it against a german and an english windows server 2016.

> This doesn't seem like much of an improvement to me. What where simple oneliners became complicated ones or in the time case even a dependency on another ps1 file. And it does exactly the same in the end?

InfluxDB or Grafana expects the time in UTC format. If you have a german based windows installation the timecode ist just wrong and you don't get any data to see in grafana. That's why i am implemented this function. 

```
Function DateTimeToUnixTimestamp([datetime]$DateTime)
{
    $utcDate = $DateTime.ToUniversalTime()
    # Convert to a Unix time without any rounding
    [uint64]$UnixTime = [double]::Parse((Get-Date -Date $utcDate -UFormat %s))
    return [uint64]$UnixTime
}

Write-Host "Current local Time:"
Get-Date

Write-Host "Time with new function:"
DateTimeToUnixTimestamp -DateTime (Get-Date)

Write-Host "Old variant:"
[int][double]::Parse((Get-Date -UFormat %s))
```

Output:
```
Current local Time:

Freitag, 1. September 2017 13:40:05
Time with new function:
1504266006
Old variant:
1504273206
```

Reply:
Hmmz, I thought they would result in the same but apparently not. I must also admit that I've never done it that way myself either. Wouldn't replacing it with a one liner like the one below be "better" than loading in an additional file? 

`$Time = [int64](([datetime]::UtcNow)-(get-date "1/1/1970")).TotalSeconds`

Granted that I've not looked at the performance impact of this, if any at all. Just trying to keep it as simple and lightweight as possible.

Reply:
the external file must be loaded anyway. So I find it makes no difference if you use the function or not. Of course, you can write the function in every file separately, but I personally consider it unnecessary.

Reply:
only in the ones where you're looking at performance counters, but granted it's better to have a unified way then making a couple of exceptions.

I still don't  understand this though 
`$Path = [System.Net.Dns]::GetHostEntry([string]"localhost").HostName.toLower()` 

you can do the same with `$Path = ($env:computername).ToLower()` which excludes the domain name, which in my mind would only muddle the metric path?

Other than that everything looks fine to me, I don't have a way to test this against other languages myself however so gonna have to trust you on that it actually works as advertised :)

Reply:
Oh that is just a personal think, I would would like to have the fqdn in the metrics. but we can revert that cause not everybody want to have it like this :-)

Reply:
sounds like an option can toggle this behavior